crates/holochain_types/src/dht_op.rs

```diff
@@ -393,7 +393,7 @@ impl DhtOp {
             }
             DhtOpType::RegisterDeletedBy => DhtOp::RegisterDeletedBy(signature, header.try_into()?),
             DhtOpType::RegisterDeletedEntryHeader => {
-                DhtOp::RegisterDeletedBy(signature, header.try_into()?)
+                DhtOp::RegisterDeletedEntryHeader(signature, header.try_into()?)
             }
```

Comment:
Ahaa! There's the crux the issue. You're not going to find the entry header in the DHT, if you only send out a Deleted by op! :)